client/src/main/java/com/alibaba/nacos/client/auth/process/HttpLoginProcessor.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 1999-2021 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacos.client.auth.process;
+
+import com.alibaba.nacos.api.PropertyKeyConst;
+import com.alibaba.nacos.api.common.Constants;
+import com.alibaba.nacos.client.auth.LoginAuthConstant;
+import com.alibaba.nacos.client.auth.LoginIdentityContext;
+import com.alibaba.nacos.client.utils.ContextPathUtil;
+import com.alibaba.nacos.common.http.HttpRestResult;
+import com.alibaba.nacos.common.http.client.NacosRestTemplate;
+import com.alibaba.nacos.common.http.param.Header;
+import com.alibaba.nacos.common.http.param.Query;
+import com.alibaba.nacos.common.utils.JacksonUtils;
+import com.alibaba.nacos.common.utils.StringUtils;
+import com.fasterxml.jackson.databind.JsonNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import static com.alibaba.nacos.client.naming.utils.UtilAndComs.HTTP;
+import static com.alibaba.nacos.client.naming.utils.UtilAndComs.webContext;
+
+public class HttpLoginProcessor implements LoginProcessor {
+    
+    private static final Logger SECURITY_LOGGER = LoggerFactory.getLogger(HttpLoginProcessor.class);
+    
+    private static final String LOGIN_URL = "/v1/auth/users/login";
+    
+    private final NacosRestTemplate nacosRestTemplate;
+    
+    private volatile LoginIdentityContext loginIdentityContext;
```

Comment:
Why cache loginIdentityContext in processor? 
Is it duplicated with AbstractClientAuthService?